fix bug #42936 - win_iis container_setting

Mapel88 opened the pull request:
State fix
- Failure / success validation is now working properly
- Server 2008 user can now use either numeric or string id

### What does this PR do?

### What issues does this PR fix or reference?

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Tests written?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Mapel88 committed:
fix bug #42936 - win_iis container_setting 

State fix
- Failure / success validation is now working properly
- Server 2008 user can now use either numeric or string id

salt-jenkins commented:
@Mapel88, thanks for your PR! By analyzing the history of the files in this pull request, we identified @m03, @trebortech and @rallytime to be potential reviewers.

Mapel88 committed:
Merge branch 'develop' into patch-2

Mapel88 committed:
fix bug #42936 - win_iis container_setting

State fix according to module changes.
Following the comments - mapping using dictionary instead function.

Mapel88 committed:
Merge branch 'develop' into patch-2

Mapel88 committed:
Removed V2 from win_iisV2

V2 was for my tests. forgot to remove it before committing.

Mapel88 commented:
To fix this bug you need to merge both win_iis module and state.
I don't know how to PR them both..
This is the updated module:
https://github.com/saltstack/salt/pull/42970
